packages/x-virtualizer/src/features/dimensions.ts

```diff
@@ -106,6 +107,34 @@ function initializeState(params: VirtualizerParams): Dimensions.State {
   };
 }
 
+const useCallbackWhenValueHoldsForTime = (timeoutMs: number) => {
+  const timeout = useTimeout();
+  const currentValueRef = React.useRef<any>(null);
+  const callbackRef = React.useRef<((value: any) => void) | undefined>(undefined);
+
+  return useLazyRef(() => {
+    return {
+      set: <V extends any>(value: V, callback?: (value: V) => void) => {
+        if (callback) {
+          callbackRef.current = callback;
+        }
+        if (value !== currentValueRef.current) {
+          timeout.clear();
+
+          timeout.start(timeoutMs, () => {
+            timeout.clear();
```

Comment:
`.clear()` is superfluous in these cases, the `Timeout` instance only holds a single timeout at once and clears stuff automatically as needed.

```suggestion
          timeout.start(timeoutMs, () => {
```